src/GameState/Marble.tsx

```diff
@@ -109,18 +107,18 @@ export const Marble = (props: MarbleProps) => {
 
   const onGestureEventHanlder = (event: any) => {
     if (event.nativeEvent.state === State.ACTIVE) {
-      if (
-        Math.abs(getYdiff(event.nativeEvent.absoluteY)) >
-        Math.abs(getXdiff(event.nativeEvent.absoluteX))
-      ) {
-        if (!isYDisabled) {
-          setIsXDisabled(true);
-        }
-      } else {
-        if (!isXDisabled) {
-          setIsYDisabled(true);
-        }
-      }
+      disableXorYdrag(event);
+    }
+  };
+
+  const disableXorYdrag = (event: any) => {
+    if (
+      Math.abs(getYdiff(event.nativeEvent.absoluteY)) >
+      Math.abs(getXdiff(event.nativeEvent.absoluteX))
+    ) {
+      !isYDisabled && setIsXDisabled(true);
+    } else {
+      !isXDisabled && setIsYDisabled(true);
```

Comment:
better but using !... && is a disguised way to use an if. I think you can find a better solution ;)